tests/unit/modules/supervisord_test.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+'''
+    :codeauthor: :email:`Jayesh Kariya <jayeshk@saltstack.com>`
+'''
+
+# Import Python Libs
+from __future__ import absolute_import
+
+# Import Salt Testing Libs
+from salttesting import TestCase, skipIf
+from salttesting.mock import (
+    MagicMock,
+    patch,
+    NO_MOCK,
+    NO_MOCK_REASON
+)
+
+from salttesting.helpers import ensure_in_syspath
+from salt.exceptions import CommandExecutionError
+
+ensure_in_syspath('../../')
+
+# Import Salt Libs
+from salt.modules import supervisord
+
+supervisord.__salt__ = {}
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
+class SupervisordTestCase(TestCase):
+    '''
+    TestCase for salt.modules.supervisord
+    '''
+    # 'start' function tests: 1
+
+    def test_start(self):
+        '''
+        Tests if it start the named service.
+        '''
+        mock_all = MagicMock(return_value={'retcode': 0, 'stdout': True})
+        mock_bin = MagicMock(return_value='/tmp/bin_env')
+        with patch.dict(supervisord.__salt__, {'cmd.run_all': mock_all,
+                                               'cmd.which_bin': mock_bin}):
+            self.assertTrue(supervisord.start())
```

Comment:
I think many kittens would stay alive if you would move MagicMock init into a private function and then do just something like:
```python
with patch.dict(supervisord.__salt__, {'cmd.run_all': _m_all(), 'cmd.which_bin': _m_bin()}):
    self.assertTrue(supervisord.start())
```
...where `_m_all()` and `_m_bin()` are private functions.